clean-modular-code/python-pep-8.md

```diff
@@ -14,237 +14,279 @@ kernelspec:
+
+print(f"The average value is: {average_value}")
 ```
 
-## Line Length
-
-PEP 8 guidelines suggest that each line of code (as well as comment lines) should be 79 characters wide or less. This is a common standard used in other languages, including **R**.
+* Line Length: Limit all lines to 79 characters for better readability.
 
 :::{tip}
 Most text editors allow you to set up guides to see how long your code is. You can then use these guides to create line breaks in your code.
 :::
 
+* Comments: Add a space after the `#` sign and capitalize the first letter of a comment:
 
-## Python PEP 8 Rules for White Space
-
-Some of the white space rules have already been discussed above. These including adding a single space after a comment `# Comment here`.
-
-There are also rules associated with spacing throughout your code. These include:
+`# This is a PEP 8 conforming comment`
 
-* **Add blank line before a single line comment (unless it is the first line of a cell in Jupyter Notebook)** Blank lines help to visually break up code. Consider reading this page, if all of the text was mashed together in one long paragraph, it would be more difficult to read. However, when you break the text up into related paragraphs, it becomes a lot easier to read.
+* White Space: Add space between sections of code to improve clarity.
+* Avoid using single-character letters that could be confused with numbers  (ie the letter `l` looks similar to the number one  `1` 
+* Add a blank line before a single-line comment (unless it is the first line of a cell in Jupyter Notebook)
 
 ```python
+a = 1 
 
-# Perform some math
-a = 1+2
-b = 3+4
-c = a+b 
+# Here is a commment
+b = 2 
 
-# Read in and plot some 
-precip_timeseries = pd.readcsv("precip-2019.csv")
-precip_timeseries.plot()
 ```
 
-The code below is more difficult to read as the spacing does not break up the text.
-
-```python
-# Perform some math and do some things 
-a=1+2
-b=3+4
-c=a+b 
-data=pd.readcsv("precip-2019.csv")
-data.plot()
-```
-
-* **Break up sections of code with white space:** As you are writing code, it's always good to consider readability and to break up sections of code accordingly. Breaking up your
+* **Break up sections of code with white space:** Breaking up your
 code becomes even more important when you start working in Jupyter Notebooks which offer individual cells where you can add Markdown and code.
 
 ```python
 # Process some data here 
-data=pd.readcsv("precip-2019.csv")
+data=pd.readcsv("pyos-data.csv")
 
 # Plot data - notice how separating code into sections makes it easier to read
 fig, ax = plot.subplots()
 data.plot(ax=ax)
 plt.show()
 ```
 
-## Summary -- PEP 8 and Python
-
-The text above provides a broad overview of some of the PEP 8 guidelines and conventions for writing **Python** code. It is not fully inclusive all of all the standards which are included in the full, online PEP 8 documentation.
+## PEP 8 naming conventions
```

Comment:
this is odd to me to have as its own top-level section after listing a bunch of pep 8 rules in the previous subsection, and it seems to be the same information as the example starting on line 179? i would combine those things and put them under a section on various types of conventions. i'll give an example ToC in the main comment